backend/de.metas.contracts/src/main/java/de/metas/contracts/interceptor/C_OrderLine.java

```diff
@@ -89,6 +113,8 @@ private void setFlatrateConditionsIdToCompensationGroup(final int flatrateCondit
 				.addNotEqualsFilter(I_C_OrderLine.COLUMNNAME_C_Flatrate_Conditions_ID, flatrateConditionsId > 0 ? flatrateConditionsId : null)
 				.create()
 				.list(I_C_OrderLine.class)
+				.stream()
+				.filter(ol -> !isOrderLineExcludedFromGroupFlatrateConditions(ol, groupId))
```

Comment:
the same groupId is loaded to many times!

Reply:
also, in case our product is excluded, can u pls check if we shall / shall not set the DYNATTR_SkipUpdatingGroupFlatrateConditions